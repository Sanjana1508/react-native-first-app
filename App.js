import React, { useState } from "react";
import { StyleSheet, View, Button, FlatList } from "react-native";
import TodoItem from "./components/TodoItem";
import TodoInput from "./components/TodoInput";

export default function App() {
  const [todos, setTodos] = useState([]);

  const [isAddMode, setIsAddMode] = useState(false);

  const addTodoHandler = (todoText) => {
    if (todoText.length === 0) {
      return;
    }
    setTodos((currentTodos) => [
      ...currentTodos,
      { uid: Math.random().toString(), value: todoText },
    ]);
    setIsAddMode(!isAddMode);
  };

  const removeGoalHandler = (goalId) => {
    setTodos((currentTodos) => {
      return currentTodos.filter((goal) => goal.uid !== goalId);
    });
  };

  const cancelAddTodoHandler = () => {
    setIsAddMode(!isAddMode);
  };

  return (
    <View style={styles.screen}>
      <Button
        title="Add New Todo"
        testID="addTodo"
        accessibilityLabel="pressToAdd"
        onPress={() => setIsAddMode(!isAddMode)}
      />
      <TodoInput
        visible={isAddMode}
        onAddTodo={addTodoHandler}
        onCancel={cancelAddTodoHandler}
      />
      <FlatList
        keyExtractor={(item, index) => item.uid}
        data={todos}
        renderItem={(itemData) => (
          <TodoItem
            id={itemData.item.uid}
            onDelete={removeGoalHandler}
            todo={itemData.item.value}
          />
        )}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  screen: {
    padding: 40,
  },
});
