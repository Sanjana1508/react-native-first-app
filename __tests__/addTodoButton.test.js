import { render, fireEvent, cleanup } from "@testing-library/react-native";
import React from "react";
import { Button } from "react-native";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

import App from "../App";

afterEach(cleanup);

describe("test Todo", () => {
  const mockFn = jest.fn();

  it("button test", () => {
    const { getByText } = render(<App />);
    const btn = getByText("Add New Todo");

    expect(btn).not.toBeNull();
  });

  it("No of buttons test", () => {
    const { getAllByText } = render(<App />);
    const btn = getAllByText("Add New Todo");

    expect(btn).toHaveLength(1);
  });

  it("test onpress", () => {
    const { getByText } = render(
      <Button title="Add New Todo" onPress={mockFn} />
    );

    fireEvent.press(getByText("Add New Todo"));

    expect(mockFn).toHaveBeenCalled();

    expect(mockFn.mock.calls.length).toBe(1);

    fireEvent.press(getByText("Add New Todo"));

    expect(mockFn.mock.calls.length).toBe(2);
  });

  it("test with testId", () => {
    const testId = "addTodo";

    const { getByTestId } = render(<App />);

    const btn = getByTestId(testId);

    expect(btn).toBeTruthy();
  });

  it("test with accessibility label", () => {
    const accessibilityLabel = "pressToAdd";

    const { getByA11yLabel } = render(<App />);

    const btn = getByA11yLabel(accessibilityLabel);

    expect(btn).toBeTruthy();
  });
  it("test button title", () => {
    const title = "Add New Todo";

    const { getByText } = render(<App />);

    const btn = getByText(title);

    expect(btn.props.children).toEqual(title);
  });

  it("test with snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toBeTruthy();
  });

  it("should render button title Add New Todo", () => {
    const tree = shallow(<App />);
    expect(tree.find("Button")).toHaveLength(1);
  });
});
