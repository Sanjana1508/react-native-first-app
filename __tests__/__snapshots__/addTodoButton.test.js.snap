// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Todo test with snapshot 1`] = `
<View
  style={
    Object {
      "padding": 40,
    }
  }
>
  <View
    accessibilityLabel="pressToAdd"
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="addTodo"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add New Todo
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Todo"
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 1,
            "margin": 10,
            "padding": 10,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "width": "60%",
          }
        }
      >
        <View
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "red",
                    },
                  ]
                }
              >
                CANCEL
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                ADD
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
